cli/tests/services.bats

```diff
@@ -1497,19 +1497,15 @@ EOF
   SCRIPT="$(cat << "EOF"
     set -euo pipefail
```

Comment:
(blocking) This hung for me locally when running a few loops with:
```sh
until ! just integ-tests services.bats -- -f daemon; do :; done
```

`overmind` had been stopped by `process-compose` was still running re-parented to pid1:
```
% pstree -ws bats
-+= 00001 root /sbin/launchd
 |-+= 23155 dcarley /nix/store/sx88hhp63s7vvhb7i0bkzk1zk65km08w-process-compose-1.9.0/bin/process-compose up -f /private/var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.a6w5cy/bats-run-kEjiGl/test/1/test/.flox/run/aarch64-darwin.test/service-config.yaml -u /private/tmp/flox.tests.3JfTAh/tmp.I1KELb4c39/run/flox.a0e7cd7a.sock -L /private/var/folders/8q/spckhr654cv4xrcv0fxsrlvc0000gn/T/nix-shell.a6w5cy/bats-run-kEjiGl/test/1/test/.flox/log/services.20240905101514228952.log --tui=false
 | \--= 23162 dcarley /nix/store/pldbvp0p2z2pa76h17g797vvk201155r-coreutils-9.5/bin/sleep infinity
 \-+= 70586 dcarley /Applications/flox-kitty.app/Contents/MacOS/applet
   \-+- 70587 dcarley /Users/dcarley/.cache/flox/remote/dcarley/shell/.flox/run/bin/kitty
     \-+= 98367 root /usr/bin/login -f -l -p dcarley /nix/store/2jwdy3fbd5qv3l6aq9pi92x9glk8ccsp-kitty-0.35.0/Applications/kitty.app/Contents/MacOS/kitten run-shell --shell /bin/zsh --shell-integration enabled --cwd=/Users/dcarley/projects/flox/flox
       \-+= 98368 dcarley -zsh
         \-+= 22799 dcarley just integ-tests services.bats -- -f daemon
           \-+- 22969 dcarley /nix/store/spyn2n8l9ca1pkqn20l40l424xmqwda4-bash-5.2p26/bin/bash /nix/store/8fwlny9ym51ymsaaaqa1qxwlcawrzh5g-flox-cli-tests/bin/flox-cli-tests --pkgdb /Users/dcarley/projects/flox/flox/pkgdb/bin/pkgdb --flox /Users/dcarley/projects/flox/flox/cli/target/debug/flox --watchdog /Users/dcarley/projects/flox/flox/cli/target/debug/flox-watchdog --input-data /Users/dcarley/projects/flox/flox/test_data/input_data --generated-data /Users/dcarley/projects/flox/flox/test_data/generated services.bats -- -f daemon
             \-+- 22976 dcarley /nix/store/spyn2n8l9ca1pkqn20l40l424xmqwda4-bash-5.2p26/bin/bash /nix/store/9c474f5ihmkambi08p7y20hmvgavs3rg-bats-1.11.0/libexec/bats-core/bats --print-output-on-failure --verbose-run --timing -f daemon services.bats
               |--- 22984 dcarley /nix/store/spyn2n8l9ca1pkqn20l40l424xmqwda4-bash-5.2p26/bin/bash /nix/store/9c474f5ihmkambi08p7y20hmvgavs3rg-bats-1.11.0/libexec/bats-core/bats --print-output-on-failure --verbose-run --timing -f daemon services.bats
               \--- 22985 dcarley /nix/store/spyn2n8l9ca1pkqn20l40l424xmqwda4-bash-5.2p26/bin/bash /nix/store/9c474f5ihmkambi08p7y20hmvgavs3rg-bats-1.11.0/libexec/bats-core/bats-format-pretty --dummy-flag -T --base-path services.bats
```

We should add the following until https://github.com/flox/flox/issues/2002 is done:
```suggestion
    set -euo pipefail
    source "${TESTS_DIR}/services/register_cleanup.sh"
```

The `"$FLOX_BIN" services stop` a few lines below (but I couldn't include in the suggestion) is arguably surplus if the watchdog is shutting down services after the activation command has finished but could be left to be explicit.